noir-projects/aztec-nr/aztec/src/note/lifecycle.nr

```diff
@@ -48,8 +47,8 @@ pub fn create_note_hash_from_public<Note, N, M>(
     note: &mut Note
 ) where Note: NoteInterface<N, M> {
     let contract_address = (*context).this_address();
-
-    let header = NoteHeader { contract_address, storage_slot, nonce: 0, is_transient: true };
+    // Public note hashes are transient, but have no side effect counters, so we just need note_hash_counter != 0
+    let header = NoteHeader { contract_address, storage_slot, nonce: 0, note_hash_counter: 1 };
```

Comment:
The header's `.note_hash_counter` is only checked for transient-ness (!= 0) when we have a public context, so I just set to 1 here.